server/service/certificate_templates.go

```diff
@@ -0,0 +1,75 @@
+package service
+
+import (
+	"context"
+	"fmt"
+	"regexp"
+	"slices"
+	"strings"
+
+	"github.com/fleetdm/fleet/v4/server/contexts/ctxerr"
+	"github.com/fleetdm/fleet/v4/server/fleet"
+	"github.com/fleetdm/fleet/v4/server/variables"
+)
+
+// Fleet variables supported in certificate template subject names.
+var fleetVarsSupportedInCertificateTemplates = []fleet.FleetVarName{
+	fleet.FleetVarHostUUID,
+	fleet.FleetVarHostHardwareSerial,
+	fleet.FleetVarHostEndUserIDPUsername,
+}
+
+func validateCertificateTemplateFleetVariables(subjectName string) error {
+	// reject any variable that doesn't use the FLEET_VAR_ prefix.
+	re := regexp.MustCompile(`\$(?:\{)?([A-Za-z_][A-Za-z0-9_]*)\}?`)
```

Comment:
I might be missing something here, but isn't a similar pattern matching logic already implemented into `variables.Find(...)`? Why is this required?